test-utils/src/lib.rs

```diff
@@ -14,3 +14,33 @@
 // limitations under the License.
 
 pub mod random;
+
+use crypto::random::distributions::uniform::SampleRange;
+use crypto::random::Rng;
+
+pub fn random_string<R: SampleRange<usize>>(rng: &mut impl Rng, range_len: R) -> String {
+    use crypto::random::distributions::{Alphanumeric, DistString};
+    if range_len.is_empty() {
+        return String::new();
+    }
+    let len = rng.gen_range(range_len);
+    Alphanumeric.sample_string(rng, len)
+}
+
+pub fn gen_text_with_non_ascii(c: u8, rng: &mut impl Rng, max_len: usize) -> Vec<u8> {
+    assert!(!c.is_ascii_alphanumeric());
```

Comment:
Every single time you call this you do something like the below first 

 ```
if c.is_ascii_alphanumeric() {
    continue;
}
```

Why the redundancy?

Reply:
Actually, this function was written by @TheQuantumPhysicist. And I also started use it for NFT. I guess, he wanted to protect us from the wrong use of the function.